spec.html

```diff
@@ -11804,6 +11799,11 @@ <h1>InitializeHostDefinedRealm ( ): either a normal completion containing ~unuse
           1. Let _thisValue_ be _global_.
         1. Set _realm_.[[GlobalObject]] to _global_.
         1. Set _realm_.[[GlobalEnv]] to NewGlobalEnvironment(_global_, _thisValue_).
+        1. Let _newContext_ be a new execution context.
+        1. Set the Function of _newContext_ to *null*.
+        1. Set the Realm of _newContext_ to _realm_.
+        1. Set the ScriptOrModule of _newContext_ to *null*.
+        1. Push _newContext_ onto the execution context stack; _newContext_ is now the running execution context.
         1. Perform ? SetDefaultGlobalBindings(_realm_).
         1. Create any host-defined global object properties on _global_.
```

Comment:
Is the only reason we can't also move these lines up because `SetDefaultGlobalBindings` might throw? If so, should we (normatively) reconsider whether those property definitions should be permitted to throw?

Reply:
> Is the only reason we can't also move these lines up because `SetDefaultGlobalBindings` might throw?

Well, the only other possible reason would be that those remaining steps need the realm EC to exist (and maybe that it be on the EC stack). That isn't the case if the global object is ordinary, but if it's a host-defined exotic object, then I suppose its relevant internal methods could be specified to access the realm EC.